lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -73,6 +78,522 @@ struct DiagRestore {
+
+  if (Mismatched) {
+    std::string Str;
+    raw_string_ostream OS(Str);
+    hlsl::PrintPSVRuntimeInfo(OS, &DMPSV, &DMPSV, &DMPSV, &DMPSV,
+                              static_cast<uint8_t>(SM->GetKind()),
+                              DM.GetEntryFunctionName().c_str(), "");
+    OS.flush();
+    std::string Str1;
+    raw_string_ostream OS1(Str1);
+    PSV.PrintPSVRuntimeInfo(OS1, static_cast<uint8_t>(PSVShaderKind::Library),
+                            "");
+    OS1.flush();
+    EmitMismatchError("PSVRuntimeInfo", Str, Str1);
+  }
+}
+
+void PSVContentVerifier::Verify() {
+  unsigned ValMajor, ValMinor;
+  DM.GetValidatorVersion(ValMajor, ValMinor);
+  unsigned PSVVersion = hlsl::GetPSVVersion(ValMajor, ValMinor);
+
+  PSVInitInfo PSVInfo(PSVVersion);
+  if (PSV.GetRuntimeInfoSize() != PSVInfo.RuntimeInfoSize()) {
+    EmitMismatchError("PSVRuntimeInfoSize",
+                      std::to_string(PSV.GetRuntimeInfoSize()),
+                      std::to_string(PSVInfo.RuntimeInfoSize()));
+    return;
+  }
+
+  if (PSV.GetBindCount() > 0 &&
+      PSV.GetResourceBindInfoSize() != PSVInfo.ResourceBindInfoSize()) {
+    EmitMismatchError("ResourceBindInfoSize",
+                      std::to_string(PSV.GetResourceBindInfoSize()),
+                      std::to_string(PSVInfo.ResourceBindInfoSize()));
+    return;
+  }
+  VerifyResources(PSVVersion);
+
+  PSVRuntimeInfo0 *PSV0 = PSV.GetPSVRuntimeInfo0();
+  PSVRuntimeInfo1 *PSV1 = PSV.GetPSVRuntimeInfo1();
+  PSVRuntimeInfo2 *PSV2 = PSV.GetPSVRuntimeInfo2();
+
+  const ShaderModel *SM = DM.GetShaderModel();
+  VerifyEntryProperties(SM, PSV0, PSV1, PSV2);
+  if (PSVVersion > 0) {
+    if (((PSV.GetSigInputElements() + PSV.GetSigOutputElements() +
+          PSV.GetSigPatchConstOrPrimElements()) > 0) &&
+        PSV.GetSignatureElementSize() != PSVInfo.SignatureElementSize()) {
+      EmitMismatchError("SignatureElementSize",
+                        std::to_string(PSV.GetSignatureElementSize()),
+                        std::to_string(PSVInfo.SignatureElementSize()));
+      return;
+    }
+    uint8_t ShaderStage = static_cast<uint8_t>(SM->GetKind());
+    if (PSV1->ShaderStage != ShaderStage) {
+      EmitMismatchError("ShaderStage", std::to_string(PSV1->ShaderStage),
+                        std::to_string(ShaderStage));
+      return;
+    }
+    if (PSV1->UsesViewID != DM.m_ShaderFlags.GetViewID())
+      EmitMismatchError("UsesViewID", std::to_string(PSV1->UsesViewID),
+                        std::to_string(DM.m_ShaderFlags.GetViewID()));
+
+    VerifySignatures(ValMajor, ValMinor);
+
+    VerifyViewIDDependence(PSV1);
+  }
+  if (PSVVersion > 1) {
+    // PSV2 only added NumThreadsX/Y/Z which verified in VerifyEntryProperties.
+  }
+  if (PSVVersion > 2) {
+    if (DM.GetEntryFunctionName() != PSV.GetEntryFunctionName())
+      EmitMismatchError("EntryFunctionName", DM.GetEntryFunctionName(),
+                        PSV.GetEntryFunctionName());
```

Comment:
Value from PSV0 and DxilModule were reversed.
```suggestion
      EmitMismatchError("EntryFunctionName", PSV.GetEntryFunctionName(),
                        DM.GetEntryFunctionName());
```

Reply:
Good catch.
Fixed.